### PR TITLE
Add logging to plugin fetch treatment method

### PR DESCRIPTION
**Pull request by deadlycoconuts**
**What this PR does / why we need it**:
This PR introduces the asynchronous logging method to the `GetTreatmentForRequest` method of the XP plugin that generates treatments. This functionality was previously left out from an earlier PR as a mistake, and this PR solely serves to fix that.

### CODE REVIEW

File: plugins/turing/runner/experiment_runner.go [hunk 1]
```diff
@@ -64,7 +66,9 @@ func (er *experimentRunner) GetTreatmentForRequest(
 	var treatment schema.SelectedTreatment
 	var switchbackWindowId *int64
 	var err error
+
 	statusCode := http.StatusBadRequest
+	filterParams := api.FetchTreatmentRequestBody{AdditionalProperties: requestParams}
```
Comment (deadlycoconuts) [inline plugins/turing/runner/experiment_runner.go:71]: Created this new variable as it's being used twice.

File: plugins/turing/runner/experiment_runner.go [hunk 2]
```diff
@@ -78,12 +82,65 @@ func (er *experimentRunner) GetTreatmentForRequest(
...
+				filterParams,
 				requestFilter,
 			)
 		}
 	}()
 
+	var lookupRequestFilters []models.SegmentFilter
+	var errorLog *monitoring.ErrorResponseLog
+
+	if er.appContext.AssignedTreatmentLogger != nil {
```
Comment (deadlycoconuts) [inline plugins/turing/runner/experiment_runner.go:94]: This block is almost a perfect copy-paste from https://github.com/caraml-dev/xp/blob/f5eb2bd3c3ce301f392a1120232748a9255ab998/treatment-service/controller/treatment.go#L67.

File: plugins/turing/go.mod [hunk 3]
```diff
@@ -19,6 +19,8 @@ require (
 	golang.org/x/oauth2 v0.0.0-20221014153046-6fdb5e3db783
 )
 
+require github.com/google/uuid v1.3.0
```
Comment (terryyylim) [inline plugins/turing/go.mod:22]: Is this intentional? - Why is this dependency not in the above `require` block?

Comment (deadlycoconuts) [reply plugins/turing/go.mod:22]: Hmm good point; not intentional but autogenerated from `go mod tidy`. Lemme poke this around 🤔

Comment (deadlycoconuts) [reply plugins/turing/go.mod:22]: Okay it seems like after manually shifting it and then running `go mod tidy`, the dependency stays in the block above now 👀

**Review by krithika369 — APPROVED** [review]
LGTM. Thanks for catching and fixing this!

**Comment by deadlycoconuts** [comment]
Thanks for the quick review; merging this now! 🚀

**Pull request merged**